interp/src/debugger/new_parser.rs

```diff
@@ -0,0 +1,99 @@
+use super::source::structures::{GroupContents, NewSourceMap};
+use crate::errors::InterpreterResult;
+use pest_consume::{match_nodes, Error, Parser};
+use std::collections::HashMap;
+type ParseResult<T> = std::result::Result<T, Error<Rule>>;
+type Node<'i> = pest_consume::Node<'i, Rule, ()>;
+
+// include the grammar file so that Cargo knows to rebuild this file on grammar changes
+const _GRAMMAR: &str = include_str!("new_metadata.pest");
+
+#[derive(Parser)]
+#[grammar = "debugger/new_metadata.pest"]
+pub struct MetadataParser;
+
+#[pest_consume::parser]
+impl MetadataParser {
+    fn EOI(_input: Node) -> ParseResult<()> {
+        Ok(())
+    }
+    pub fn num(input: Node) -> ParseResult<u64> {
+        input
+            .as_str()
+            .parse::<u64>()
+            .map_err(|_| input.error("Expected number"))
```

Comment:
You can use `match_nodes!` here to avoid needing to invoke `parse`. If we've
reached this point it should already be parseable anyway so no need for this
failure path.

See
https://github.com/calyxir/calyx/blob/1be1523daef5d0a3354c7866f8ecb71810b6e1ac/interp/src/debugger/parser/command_parser.rs
for an example